vizro-core/changelog.d/20241119_173819_huong_li_nguyen_remove_variables.md

```diff
@@ -0,0 +1,48 @@
+<!--
+A new scriv changelog fragment.
+
+Uncomment the section that is right (remove the HTML comment wrapper).
+-->
+
+<!--
+### Highlights ✨
+
+- A bullet item for the Highlights ✨ category with a link to the relevant PR at the end of your entry, e.g. Enable feature XXX. ([#1](https://github.com/mckinsey/vizro/pull/1))
+
+-->
+
+### Removed
+
+- Removed all CSS variables from `variables.css` and `token_names.css`, replacing them with CSS variables from `vizro-bootstrap.min.css`.
+  Refer to [`vizro-bootstrap.min.css](https://github.com/mckinsey/vizro/blob/main/vizro-core/src/vizro/static/css/vizro-bootstrap.min.css) for the updated CSS variables. ([#886](https://github.com/mckinsey/vizro/pull/886))
```

Comment:
Good question—ideally, if we had a fully developed Bootstrap theme, we should source the variables directly from Bootstrap's own variables. Unfortunately, that's not the case right now, so using Bootstrap variables won't be very helpful, as not all our tokens are mapped to selected Bootstrap variables. I am also not sure at this moment whether this will actually be possible because we have a lot more tokens than available default Bootstrap variables. So there might always be a case where you need to check the names of the underlying tokens.

In practice, you simply need to place this file in your assets folder, and your code configurator will autocomplete the values for you. However, keep in mind that this change will only impact those who have used our old token names and now need to switch to our new token names. The names themselves remain the same; they are just written differently e.g. `border-subtle-alpha-01` to `border-subtleAlpha01`

Reply:
Makes sense. The user I was thinking of here isn't someone who have used our old token names and need to switch. It's someone who wants to customise their CSS for the first time e.g. make a new component that follows our colour scheme and works in both light and dark theme. AFAICT at the moment the relevant bit of docs is https://vizro.readthedocs.io/en/stable/pages/user-guides/custom-css/#change-the-styling-of-a-container where we do `var(--surfaces-bg-card)` (is that one still valid?). But it's not really explained what the `var` is for or why you should use that.

Maybe it would be worth a note in the docs saying that where possible you should use `var` to make sure CSS is responsive to theme changes and point to a list of what variable names we do have. This could just be a link to the unminified version of this file or to the bootstrap docs with a disclaimer that we don't use all their variable names. e.g. a bit like the tip here that links to the readable version of the stylesheet. https://hellodash.pythonanywhere.com/adding-themes/dcc-components

This group of users isn't a big group at all so not worth worrying about much but I feel like there should probably be something to help them find variable names in the docs (not in the changelog, I just happened to put the comment here).

Reply:
P.S. did you see this cheatsheet before? It's really good. https://dashcheatsheet.pythonanywhere.com/

Reply:
I've added this in now: https://github.com/mckinsey/vizro/pull/886/commits/93587424d0b1b8e7db9957dbf4dbf4e5f46a14de :)

Right now, I'd suggest not using the unminified CSS version to look up variables. It just has way more info than you need and isn't great for quick references. I can explain more about this later.

Also, I updated the example to skip using one of our tokens. Your comment made me rethink it, and I realized people probably wouldn't use it that way.

@stichbury - could you take a look at the docs changes in that commit? Thank you 🙏